File: .cursor/hooks/notion_gui_popup.mjs
/**
 * Popup Notion local workflow GUI when prompt contains command keywords.
 */
import { readFileSync } from "node:fs";
import { resolve } from "node:path";
import { spawn } from "node:child_process";

const KEYWORDS = [
  "notion菜单",
  "notion 面板",
  "打开notion面板",
  "打开notion菜单",
  "同步notion",
  "notion同步",
  "notion向导",
  "notion crud",
  "notion增删改查",
];

function readPayload() {
  const raw = readFileSync(0, "utf8");
  if (!raw.trim()) return {};
  try {
    return JSON.parse(raw);
  } catch {
    return { raw };
  }
}

function extractPrompt(payload) {
  for (const key of ["prompt", "userPrompt", "text", "input"]) {
    if (typeof payload[key] === "string") return payload[key];
  }
  try {
    return JSON.stringify(payload);
  } catch {
    return "";
  }
}

function shouldPopup(promptText) {
  const lower = String(promptText || "").toLowerCase();
  return KEYWORDS.some((k) => lower.includes(k));
}

function openGui(projectDir) {
  const script = resolve(projectDir, ".cursor", "tools", "notion_gui_menu.ps1");
  const child = spawn(
    "powershell",
    ["-NoProfile", "-ExecutionPolicy", "Bypass", "-File", script],
    {
      cwd: projectDir,
      detached: true,
      stdio: "ignore",
      windowsHide: false,
    }
  );
  child.unref();
}

function main() {
  const payload = readPayload();
  const prompt = extractPrompt(payload);
  if (shouldPopup(prompt)) {
    const projectDir = process.env.CURSOR_PROJECT_DIR || process.cwd();
    openGui(projectDir);
  }
  process.stdout.write(`${JSON.stringify({ permission: "allow" })}\n`);
}

main();

File: .cursor/hooks/notion_gui_popup.py
"""Popup Notion local workflow GUI when prompt contains command keywords."""

from __future__ import annotations

import json
import os
import subprocess
import sys
from pathlib import Path

KEYWORDS = (
    "notion菜单",
    "notion 面板",
    "打开notion面板",
    "打开notion菜单",
    "同步notion",
    "notion同步",
    "notion向导",
    "notion crud",
    "notion增删改查",
)


def read_payload() -> dict:
    raw = sys.stdin.read()
    if not raw.strip():
        return {}
    try:
        return json.loads(raw)
    except Exception:
        return {"raw": raw}


def extract_prompt(payload: dict) -> str:
    for key in ("prompt", "userPrompt", "text", "input"):
        value = payload.get(key)
        if isinstance(value, str):
            return value
    return json.dumps(payload, ensure_ascii=False)


def should_popup(text: str) -> bool:
    lower = text.lower()
    return any(k in lower for k in KEYWORDS)


def open_gui(project_dir: Path) -> None:
    script = project_dir / ".cursor" / "tools" / "notion_gui_menu.ps1"
    if not script.exists():
        return
    flags = 0
    if os.name == "nt":
        flags = 0x00000008  # DETACHED_PROCESS
    subprocess.Popen(
        [
            "powershell",
            "-NoProfile",
            "-ExecutionPolicy",
            "Bypass",
            "-File",
            str(script),
        ],
        cwd=str(project_dir),
        creationflags=flags,
        stdout=subprocess.DEVNULL,
        stderr=subprocess.DEVNULL,
    )


def main() -> None:
    payload = read_payload()
    prompt = extract_prompt(payload)
    if should_popup(prompt):
        project_dir = Path(os.environ.get("CURSOR_PROJECT_DIR") or os.getcwd())
        open_gui(project_dir)
    sys.stdout.write(json.dumps({"permission": "allow"}, ensure_ascii=False) + "\n")
    sys.stdout.flush()


if __name__ == "__main__":
    main()
